pkg/manager/internal.go

```diff
@@ -439,6 +439,11 @@ func (cm *controllerManager) Start(ctx context.Context) (err error) {
 		return fmt.Errorf("failed to start other runnables: %w", err)
 	}
 
+	// Start and wait for sources to start.
```

Comment:
```suggestion
	// Start WarmupRunnables and wait for warmup to complete.
```

Reply:
Addressed in https://github.com/kubernetes-sigs/controller-runtime/pull/3192/commits/6bb4616ba3e651c98015a5382549982074f563ac